src/components/browse/sidebar/sidebar.tsx

```diff
@@ -1,40 +1,76 @@
-import React from 'react';
-import Search from './searchbar'
-import Filter from './filter';
-import { Row, Col, Button } from 'react-bootstrap'
-import { facetModel } from '../../../models/facets'
+import React, { Dispatch, SetStateAction, useState } from "react";
+import Search from "./searchbar";
+import Filter from "./filter";
+import { Row, Col, Button } from "react-bootstrap";
+import { facetModel, facetFilterModel } from "../../../models/facets";
+import { searchResponseModel } from "../../../models/dataset";
+import { getDatasetsSearchResp } from "../../../api/browse";
 
 interface sidebarProps {
   facetList: facetModel[] | null;
-};
+  setSearchResp: Dispatch<SetStateAction<searchResponseModel | null>>;
+  searchKeyword: string
+  limit: number
+  setSearchKeyword: Dispatch<SetStateAction<string>>
+  setFilterDict: Dispatch<SetStateAction<facetFilterModel[]>>;
+  filterDict: facetFilterModel[];
+}
 
 const Sidebar = (props: sidebarProps) => {
+  const [check, setCheck] = useState<Map<string, boolean>>(new Map<string, boolean>());
+  const handleClear = () => {
+    getDatasetsSearchResp(props.setSearchResp, [], '*', 0, props.limit);
+    check.forEach((value: boolean, key: string) => {
+      setCheck(check.set(key, false))
+    })
+    props.setFilterDict([])
+  };
+
+  const handleFilter = () => {
+    getDatasetsSearchResp(props.setSearchResp, props.filterDict, props.searchKeyword, 0, props.limit);
+  };
+
   return (
     <div>
       <Row>
-        <Search />
+        <Search searchKeyword={props.searchKeyword}
+          setSearchKeyword={props.setSearchKeyword}
+          setSearchResp={props.setSearchResp}
+          limit={props.limit} />
       </Row>
       {props.facetList === null || props.facetList.length < 1 ? null : (
-        <div className='bg-light border p-2 rounded-3'>
+        <div className="bg-light border p-2 rounded-3">
           <Row>
-            {
-              props.facetList.map((facet, index) => (
-                <Filter facet={facet} />
-              ))
-            }
+            {props.facetList.map((facet, index) => (
+              <Filter
+                facet={facet}
+                key={index}
+                check={check}
+                setCheck={setCheck}
+                setFilterDict={props.setFilterDict}
+                searchKeyword={props.searchKeyword}
+                filterDict={props.filterDict}
+              />
+            ))}
           </Row>
-          <Row className="mb-2">
-            <Col className="offset-2" xs md lg={4}>
-              <Button className='btn-warning w-100'>Clear</Button>
-            </Col>
-            <Col xs md lg={6} className="">
-              <Button className='btn-success w-100'>Filter</Button>
-            </Col>
-          </Row>
-        </div>
-      )}
+        </div>)}
+      <div className="bg-light border p-2 rounded-3">
```

Comment:
Why are the buttons in their own divs? Is this from the wireframe?

Reply:
The buttons are not in their own divs. There are 2 divs in total, with the searchbar in one div and the Filter facets, clear and filter button in another which is contained within the first div.

Reply:
Okay. Move the entire div for the buttons to be between lines 55 and 56, and remove the classes from their div.